disttask/framework/scheduler/manager.go

```diff
@@ -0,0 +1,312 @@
+// Copyright 2023 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package scheduler
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/tidb/disttask/framework/proto"
+	"github.com/pingcap/tidb/resourcemanager/pool/spool"
+	"github.com/pingcap/tidb/resourcemanager/util"
+	"github.com/pingcap/tidb/util/logutil"
+	"go.uber.org/zap"
+)
+
+var (
+	schedulerPoolSize       int32 = 4
+	subtaskExecutorPoolSize int32 = 10
+	checkTime                     = time.Second
+)
+
+var (
+	// newPool is used to create a new Pool.
+	// use this variable to mock the Pool in unit test.
+	newPool = func(name string, size int32, component util.Component, options ...spool.Option) (Pool, error) {
+		return spool.NewPool(name, size, component, options...)
+	}
+	// newScheduler is used to create a new InternalScheduler.
+	// use this variable to mock the InternalScheduler in unit test.
+	newScheduler = NewInternalScheduler
```

Comment:
Is it better to introducer a `ManagerBuilder` with methods `SetSchedulerFactory(Scheduler func())` to build `Manager` into different scenes. Static variable may introduces some troubles, ie, to make test unstable if tests are running in concurrency.